### PR TITLE
flash: Move dependency on FLASH_PAGE_LAYOUT where it belongs

### PR DESCRIPTION
**Pull request by de-nordic**
The commit adds dependency on Kconfig FLASH_PAGE_LAYOUT to subsystems that really require it:
 FCB, NVS, LittleFS
and removes direct selection from '*.conf' files where no longer needed.

### CODE REVIEW

**Review by jori-nordic — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

File: drivers/flash/Kconfig.stm32 [hunk 1]
```diff
@@ -11,7 +11,6 @@ config SOC_FLASH_STM32
 	depends on DT_HAS_ST_STM32_FLASH_CONTROLLER_ENABLED
 	select FLASH_HAS_DRIVER_ENABLED
 	default y
-	select FLASH_PAGE_LAYOUT
```
Comment (erwango) [inline drivers/flash/Kconfig.stm32:14]: This is also done in few other files such as: 
./drivers/flash/Kconfig.sam
./drivers/flash/Kconfig.stm32_ospi
./drivers/flash/Kconfig.stm32_qspi

Do you want to clean them as well ?

Comment (de-nordic) [reply drivers/flash/Kconfig.stm32:14]: Yes. I have missed them.

Comment (de-nordic) [reply drivers/flash/Kconfig.stm32:14]: Fixed.

**Review by jori-nordic — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by nashif** [comment]
`treewide` is not an area/subsystem and should not be used in commit messages, this is flash, so use "flash: ...."

**Comment by de-nordic** [comment]
> `treewide` is not an area/subsystem and should not be used in commit messages, this is flash, so use "flash: ...."

Will fix.

**Review by kartben — APPROVED** [review]

**Pull request merged**